### DISCUSSION TITLE
standalone widget iframe

### DISCUSSION THREAD

**Discussion by epertinez** [filamentphp/filament#8905] [General]
Hi,
I did a very useful graph in a filament page with a widget, but now I would like to create a standalone page that only returns the widget so I can embed it to another page using iframe.
Authorization shouldn't be a problem as somehow the user is logged in when I insert the widget in the iframe.

Now... is there any easy way to create an standalone page with its unique route for a widget? Could someone guide me?

Thanks

**Comment by leandrocfe** [comment]
[this setup](https://filamentphp.com/docs/3.x/widgets/installation) ?

**Reply by epertinez** [reply]
Perfect. That is what I needed. Css formating did not seem to work, but I did not play with vita and all this css compiling, so it might be it.
A couple of things:

a) You don't need to create a view or a controller for your Widget. You simply route it from web.php

```
use App\Filament\Widgets\MyWidget;

Route::get('/theUrlYouWanna/{someVariable}', MyWidget::class);

```
b) By some magic, if the widget gets to be routed to... looks for resources/views/components/layouts/app.blade.php (you have to create it) and wraps itself around before viewing... clever.

c) By some more magic, your variables become automagically set in tour widget. You only have to create a public variable with the same name than in the route...

```
class MyWidget extends ChartWidget
{  public $someVariable; 

```
... and that's all! The variable gets the value from the route.

Too much magic for an old school programmer like me... but clean code.
